### PR TITLE
Get `use VERSION` to import the corresponding `builtin` bundle

### PR DESCRIPTION
**Pull request by leonerd**
A few months ago we added `use builtin :VER` bundles, specifically with the plan that we'd eventually get `use VERSION` to imply it. This PR finally completes the last step of the process, allowing a simple `use VERSION` declaration to import all of the (stable) builtin symbols at that version.

Now the full user convenience is

```perl
use v5.40;
say "True is ", true;
weaken( my $weakref = [] );
# etc...
```

This also applies to the `-E` commandline switch, allowing short oneliners like

```
$ perl -E 'say reftype []'
ARRAY
```

### CODE REVIEW

File: pod/perlfunc.pod
Comment (Grinnz) [inline pod/perlfunc.pod]: perhaps "any lexically imported effects" or "any imported effects from use VERSION" so as not to be misleading

File: t/comp/use.t
Comment (happy-barney) [inline t/comp/use.t]: shouldn't this test also availability of symbols ?

Comment (happy-barney) [reply t/comp/use.t]: also it would be nice to test behaviour when code already has exported symbol, eg:
```
use Scalar::Util qw (weaken);
use v5.40;
```
(and vice versa)

Comment (leonerd) [reply t/comp/use.t]: Symbol visibility is tested by a different test; see the one on line 181.

Comment (happy-barney) [reply t/comp/use.t]: can you move them closer?

also related test:
```
eval "{ use v5.39 } use v5.20; true";
ok $@, ...
```

File: pod/perlfunc.pod
Comment (leonerd) [reply pod/perlfunc.pod]: Ahyes; reworded in 3fadae8c81 (will fixup before merge)

**Review by tonycoz — APPROVED** [review]

**Comment by leonerd** [comment]
Having discussed this with PSC today, @haarg expressed concern about the new restriction on downgrading version declarations. I won't merge this yet then, and see what can be done differently in order to remove that restriction.

**Comment by leonerd** [comment]
Logic rewritten to perform some unexport of symbols at `use VERSION` time, removing the restriction on downgrading. PTAL.

Once merged this really needs a perldelta, which I'll write in a separate commit.

**Comment by haarg** [comment]
Looks good to me.

**Comment by leonerd** [comment]
@tonycoz - I rewrote quite a lot of this since your last review. This still good with you?

File: builtin.c
Comment (tonycoz) [inline builtin.c:712]: Does the removal of the SvREFCNT_dec() fix some other problem?

If so the commit message should mention it.

I suspect this was resulting in freeing of a freed scalar further down the line but the commit message should mention it if so.

**Comment by tonycoz** [comment]
> @tonycoz - I rewrote quite a lot of this since your last review. This still good with you?

Looks fine except for wondering about that SvREFCNT_dec() removal.

File: builtin.c
Comment (leonerd) [reply builtin.c:712]: Actually it was indirectly *causing* a problem. Nothing ought to be looking in the pad at the SVs in tombstone entires, but if anything did then previously they'd see the address of an SV that has longsince been freed (or perhaps more likely reused). By not freeing it, we instead leave it as a simple SVt_NULL. I thought that was safer than freeing it then NULL'ing it out, in case something tried to dereference it anyway.

Comment (leonerd) [reply builtin.c:712]: I've now made some further adjustments to the code in `pad.c`, and expanded on the wording in the commit message to explain the situation.

**Review by tonycoz — APPROVED** [review]

**Pull request merged**
